### PR TITLE
Propagate ImagePullBackOff and ErrImagePull in Build status

### PR DESCRIPTION
**Pull request by tylerphelan**

### CODE REVIEW

**Comment by tylerphelan** [comment]
Old Build status when a pod is ImagePullBackOff or ErrImagePull:

Screen Shot 2020-07-31 at 10 42 07 AM

New Build status:

Screen Shot 2020-07-31 at 10 40 15 AM

File: go.sum [hunk 1]
```diff
@@ -1803,3 +1803,4 @@ sourcegraph.com/sqs/pbtypes v0.0.0-20180604144634-d3ebe8f20ae4/go.mod h1:ketZ/q3
 sourcegraph.com/sqs/pbtypes v1.0.0/go.mod h1:3AciMUv4qUuRHRHhOG4TZOB+72GdPVz5k+c648qsFS4=
 vbom.ml/util v0.0.0-20160121211510-db5cfe13f5cc/go.mod h1:so/NYdZXCz+E3ZpW0uAoCj6uzU2+8OWDFv/HxUSs7kI=
 vbom.ml/util v0.0.0-20180919145318-efcd4e0f9787/go.mod h1:so/NYdZXCz+E3ZpW0uAoCj6uzU2+8OWDFv/HxUSs7kI=
+
```
Comment (sukhil-suresh) [inline go.sum:1806]: no actual change to go.sum file

File: pkg/reconciler/build/build.go [hunk 2]
```diff
@@ -158,6 +162,22 @@ func conditionForPod(pod *corev1.Pod) corev1alpha1.Conditions {
 				LastTransitionTime: corev1alpha1.VolatileTime{Inner: metav1.Now()},
 			},
 		}
+	case corev1.PodPending:
+		for _, c := range pod.Status.InitContainerStatuses {
+			if c.State.Waiting != nil &&
+				(c.State.Waiting.Reason == errImagePullMsg || c.State.Waiting.Reason == imagePullBackOffMsg) {
```
Comment (matthewmcnew) [inline pkg/reconciler/build/build.go:168]: Should we just always bubble up reason from a waiting container?

Comment (tylerphelan) [reply pkg/reconciler/build/build.go:168]: @matthewmcnew which container's reason? the pod's reason?

**Comment by tylerphelan** [comment]
This passes any waiting container's status Reason and Message to Build status

**Pull request merged**